tools/flaky-test-reporter/ghutil/ghutil.go

```diff
@@ -0,0 +1,327 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// ghutil.go provides generic functions supporting Github operations
+// It helps overcome Github rate limit errors
+
+package ghutil
+
+import (
+	"context"
+	"io/ioutil"
+	"strings"
+	"log"
+	"fmt"
+	"time"
+
+	"golang.org/x/oauth2"
+	"github.com/google/go-github/github"
+)
+
+const (
+	maxRetryCount = 5
+	tokenReserve  = 50
+
+	// IssueOpenState is the state of open github issue
+	IssueOpenState IssueStateEnum = "open"
+	// IssueCloseState is the state of closed github issue
+	IssueCloseState IssueStateEnum = "close"
+	// IssueAllState can be used as filter for querying genericList including closed genericList
+	IssueAllState  IssueStateEnum = "all"
+)
+
+// IssueStateEnum represents different states of Github Issues
+type IssueStateEnum string
+
+// client for github authentication
+// without explicitly authenticate this client can still be used for anonymous operations
+var client *github.Client
+var ctx = context.Background()
+
+// Authenticate explicitly authenticate github with giving token
+func Authenticate(tokenFilePath *string) error {
+	b, err := ioutil.ReadFile(*tokenFilePath)
+	if err != nil {
+		return err
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: strings.TrimSpace(string(b))},
+	)
+	client = github.NewClient(oauth2.NewClient(ctx, ts))
+	return nil
+}
+
+// GetCurrentUser gets authenticated user
+func GetCurrentUser() (*github.User, error) {
+	var res *github.User
+	_, err := retry(
+		fmt.Sprintf("getting current user"),
```

Comment:
Unnecessary Sprintf, use the raw string.

Reply:
fixed